docs/docs/icechunk-python/cheatsheets/git-users.md

```diff
@@ -0,0 +1,167 @@
+
+## Working with branches
+
+Icechunk has [branches](../version-control.md#branches) similar to git.
+
+### Creating a branch
+
+In practice, this means the workflow is different from git. For instance, I wanted to make a new branch based on the `main` branch on my existing git repository and then commit my changes in git this is how I would do it:
+
+```bash
+# Assume currently on main branch
+# create branch
+git checkout -b my-new-branch
+# stage changes
+git add myfile.txt
+# commit changes
+git commit -m "My new branch"
+# push to remote
+git push origin -u my-new-branch
+```
+
+In Icechunk, you would do the following:
+
+```python
+# We create the branch
+repo.create_branch("my-new-branch", repo.lookup_branch("main"))
+# create a writable session
+session = repo.writable_session("my-new-branch")
+...  # make some changes
+# commit the changes
+session.commit("My new branch")
+```
+
+Two things to note:
+
+1. When we create a branch, the branch is now available for any other instance of this `Repository` object. It is not a local branch, it is created in the repositories storage backend.
+2. When we commit the changes are immediately visible to other users of the repository. There is not concept of a local commit, all snapshots happen in the storage backend.
+
+### Checking out a branch
+
+In git, you can check out a branch by using the `git checkout` command. Icechunk does not have the concept of checking out a branch, instead you create [`Session`s](../reference/#icechunk.Session) that are based on the tip of a branch.
+
+We can either check out a branch for [read-only access](../reference/#icechunk.Repository.readonly_session) or for [read-write access](../reference/#icechunk.Repository.writable_session).
+
+```python
+# check out a branch for read-only access
+session = repo.readonly_session("my-new-branch")
+# check out a branch for read-write access
+session = repo.writable_session("my-new-branch")
+```
+
+Once we have checked out a session, the [`store`](../reference/#icechunk.Session.store) method will return a [`Store`](../reference/#icechunk.Store) object that we can use to read and write data to the repository with `zarr`.
+
+### Resetting a branch
+
+In git, you can reset a branch to previous commit. Similarly, in Icechunk you can [reset a branch to a previous snapshot](../reference/#icechunk.Repository.reset_branch).
+
+```python
+# reset the branch to the previous snapshot
+repo.reset_branch("my-new-branch", "198273178639187")
+```
+
+!!! warning
+    This is a destructive operation. It will overwrite the branch with the snapshot immediately. It can only be undone by resetting the branch again.
```

Comment:
```suggestion
    This is a destructive operation. It will overwrite the branch reference with the snapshot immediately. It can only be undone by resetting the branch again.
```